### PR TITLE
Bugfix/875 rar oppforsel i filter i api delegering

### PR DESCRIPTION
**Pull request by Albertlarsen**
## Related Issue(s)
- #875 
- #861 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/components/Filter/OptionDisplay.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export const OptionDisplay = ({
     return (
       <button
         className={classes.option}
-        key={option.value}
+        key={option.label}
```
Comment (Albertlarsen) [inline src/components/Filter/OptionDisplay.tsx:92]: because resource register returns both Arbeids and velferdsetaten and Nav (Arbeids og velferdsetaten)

Comment (allinox) [reply src/components/Filter/OptionDisplay.tsx:92]: Does that mean we will now have both Arbeids and velferdsetaten and Nav (Arbeids og velferdsetaten) in our display?
That sounds suboptimal...

**Review by allinox — CHANGES_REQUESTED** [review]
I still don't understand what caused that "kasse" situation in the bug report, but I can't trigger it when I run this branch locally, so I guess it's fixed somehow..? 😅

File: src/components/Filter/OptionDisplay.tsx (hunk 1 above)
Comment (Albertlarsen) [reply src/components/Filter/OptionDisplay.tsx:92]: I don't see any other solution than checking if 2 resourceOwners have the same org code and then only return one of them (We could then add a rule for which of them to return in those scenarios but I don't know what rule we should use for which of them to use, maybe we should huddle about it tomorrow?)

**Comment by Albertlarsen** [comment]
@allinox Sorry, I should have said something about it. The reason of the weird Kasse error was because there where duplicate keys in react. "Arbeids and velferdsetaten" and "Nav (Arbeids og velferdsetaten)" had the same key since they had the same org code. Duplicate keys in react causes unexpected errors and this was one of them (Seemed like everytime you searched for something "Arbeids and Velferdsetaten" was automatically added as a duplicate and also always displayed, maybe because it didn't actually exist but was just a duplicate). So duplicate keys caused more duplicate keys and so on so everytime you searched for something, more duplicate keys was added and more duplicate FilterOptions were added.

File: src/features/singleRight/components/SearchSection/SearchSection.tsx [hunk 2]
```diff
@@ -272,7 +272,6 @@ export const SearchSection = ({ onAdd, onUndo }: SearchSectionParams) => {
           values={filters}
           onApply={(filters) => {
             setFilters(filters);
-            setCurrentPage(1);
```
[the remark this answers is not in the document]
Comment (Albertlarsen) [reply src/features/singleRight/components/SearchSection/SearchSection.tsx:275]: Good and well thought through comment :)

File: src/components/Filter/OptionDisplay.tsx (hunk 1 above)
Comment (allinox) [reply src/components/Filter/OptionDisplay.tsx:92]: Yeah, no I don't think we should try and fix every issue that isn't our fault with special case code.
That being said though,, I thought the way the code was written in the BFF was that we filtered out the orgs from the list of resources, based on the orgcode or orgnumber? So shouldn't that have meant that we wouldn't get duplicates like these?
(I haven't actually cheched right now though, so I might be wrong)

**Comment by allinox** [comment]
@Albertlarsen 
Aha! That makes sense!
Thanks for the explanation and kudos for figuring it out! 🙌

**Review by allinox — APPROVED** [review]

**Pull request merged**